src/librustc_mir/transform/check_consts/mod.rs

```diff
@@ -34,6 +34,7 @@ impl Item<'mir, 'tcx> {
         def_id: DefId,
         body: mir::ReadOnlyBodyAndCache<'mir, 'tcx>,
     ) -> Self {
+        // FIXME: pass down the param env from the caller
```

Comment:
It should be correct to get it here, no?

Reply:
yes, but we also compute it outside, seemed like a good situation to remove an unnecessary query call.

Reply:
What if the caller got the `ParamEnv` from this `Item` struct instead?

Reply:
done